runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/kotlin/kotlinClasspathUtils.kt

```diff
@@ -23,7 +28,9 @@ internal fun Project.classpathOf(sourceSet: KotlinSourceSet): FileCollection {
              * Dokka needs only HMPP variant
              * Ignore [org.jetbrains.kotlin.gradle.plugin.mpp.KotlinCommonCompilation]  for `commonMain`  sourceSet with name `main`
              */
-            .filterNot { compilation -> isHMPPEnabled && compilation is KotlinCommonCompilation && compilation.name == "main" }
+            .filterNot { compilation ->
+                shouldKeepBackwardsCompatibility && isHMPPEnabled && compilation is KotlinCommonCompilation && compilation.name == "main"
```

Comment:
this is fix for #3038 (single target MPP projects)
Not sure about the real reason, but while debugging in my own project, I see that `isHMPPEnabled=true`, even while I have no `kotlin.mpp.enableGranularSourceSetsMetadata` specified anywhere in project.
BTW, comment is a little misleading also, as it mentions `Compatibility Metadata Variant` which is enabled by other flag `kotlin.mpp.enableCompatibilityMetadataVariant=true` - may be checking for correct flag will be enough to keep compatibility, but haven't tested it. Also, PR introduced this changes also don't have description (https://github.com/Kotlin/dokka/pull/2634) so I'm a little afraid on how those changes will affect building of other projects

Reply:
(in reply to the [comment](https://github.com/Kotlin/dokka/pull/3081/files#r1269975175))

Yeah, this fix is quite janky - we badly needed a solution that worked (as it also blocked some projects), and this is what we ended up with after discussing it with the Kotlin Multiplatform team. I don't fully understand it though, and I certainly didn't expect it to backfire...

> I'm a little afraid on how those changes will affect building of other projects

This is where it becomes interesting :)) The reason we put it there to begin with is because we couldn't build ktor without it. I've ran your branch on latest ktor, and indeed it doesn't work :( [stacktrace here](https://gist.github.com/IgnatBeresnev/27c623e531ebc822b033094020900ca1)

However, this particular filter is the reason for why single-target multiplatform projects don't work: KGP 1.9.0 removed a redundant commonMain compilation, and we filter out the only thing that's left, so compilations end up being empty -> reduce on an empty collection.

I've already found a fix for this particular problem, and it seems to work in both cases, so I'll create a separate PR, and can tag you in it if you're curious :)

Reply:
Yeah, sound good, Thanks!